staging/src/k8s.io/client-go/util/csaupgrade/upgrade.go

```diff
@@ -107,21 +134,76 @@ func UpgradeManagedFields(
 			entry.Subresource == "")
 	})
 
-	// Wipe out last-applied-configuration annotation if it exists
-	annotations := accessor.GetAnnotations()
-	delete(annotations, csaAnnotationName)
+	return filteredManagers, nil
+}
 
-	// Commit changes to object
-	accessor.SetAnnotations(annotations)
-	accessor.SetManagedFields(filteredManagers)
+// Ensures a given manager exists that owns the given field paths
+// If there is an entry for the given manager already, inserts
+//
+//	field path into its ownership set.
+//
+// Otherwise, creates manager at latest APIVersion which owns the field
+// Returns an error if there was an issue with decoding/encoding. Or nil if
+// the given field path are now owned by the desired manager
```

Comment:
I'd really rather avoid doing that if we can, playing with managed fields like this is getting more and more tedious :-/

Reply:
We should also probably unit-test this function if we end-up keeping it, but let's wait for now.